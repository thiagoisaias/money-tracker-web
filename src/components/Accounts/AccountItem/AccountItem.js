import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { devices } from "../../../utils/devices";

const Container = styled.div`
  margin: 16px 0;
  padding: 6px 0;
  border-radius: 2px;
  height: ${props => (props.isActive ? "56px" : "36px")};
  transition: ease 0.25s;
  overflow: hidden;

  &:hover {
    border-color: #ddd;
    cursor: pointer;
  }

  @media ${devices.small} {
    border-top: 1px solid #f2f2f2;
    border-bottom: 1px solid #f2f2f2;
  }

  @media ${devices.mediumUp} {
    border: 1px solid #f2f2f2;
    padding-left: 16px;
  }
`;

const Name = styled.div`
  margin-bottom: 8px;
`;

const Balance = styled.div`
  color: #67b8ff;
  font-weight: 600;
`;

const ActionsContainer = styled.div`
  display: flex;
  color: #777;
  font-size: 12px;
  margin-top: 8px;
`;

const Action = styled.div`
  margin-right: 16px;
  &:hover {
    font-weight: 600;
    cursor: pointer;
  }
`;

const AccountItem = props => {
  const {
    accountData,
    handleActiveItem,
    handleDelete,
    handleEdit,
    isActive
  } = props;

  return (
    <Container
      isActive={isActive}
      onClick={() => handleActiveItem(accountData.id)}
    >
      <Name>{accountData.name}</Name>
      <Balance>{`$ ${accountData.initialBalance.toLocaleString()}`}</Balance>
      <ActionsContainer>
        <Action
          onClick={() => {
            handleEdit();
          }}
        >
          {"Edit"}
        </Action>
        <Action
          onClick={() => {
            handleDelete();
          }}
        >
          {"Delete"}
        </Action>
      </ActionsContainer>
    </Container>
  );
};

AccountItem.propTypes = {
  accountData: PropTypes.shape({
    id: PropTypes.number.isRequired,
    initialBalance: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired,
  handleActiveItem: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleEdit: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired
};

export default AccountItem;
