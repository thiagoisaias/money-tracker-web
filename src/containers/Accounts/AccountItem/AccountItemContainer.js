import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import {
  deleteAccount,
  setAccountToEdit
} from "../../../store/actions/accounts/accounts";

import AccountItem from "../../../components/Accounts/AccountItem/AccountItem";

export class AccountItemContainer extends Component {
  handleEdit = () => {
    const { accountData, history, onSetAccountToEdit } = this.props;
    onSetAccountToEdit(accountData);
    history.push(`/accounts/${accountData.id}/edit`);
  };

  handleDelete = () => {
    const { accountData, onDeleteAccount } = this.props;
    if (!window.confirm("Are you sure you want to delete this account?")) {
      return;
    }
    onDeleteAccount(accountData.id);
  };

  render() {
    const {
      accountData,
      handleActiveItem,
      isActive
    } = this.props;
    return (
      <AccountItem
        accountData={accountData}
        handleActiveItem={handleActiveItem}
        handleDelete={this.handleDelete}
        handleEdit={this.handleEdit}
        isActive={isActive}
      />
    );
  }
}

AccountItemContainer.propTypes = {
  accountData: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    initialBalance: PropTypes.string.isRequired
  }).isRequired,
  handleActiveItem: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
  isActive: PropTypes.bool.isRequired,
  onDeleteAccount: PropTypes.func.isRequired,
  onSetAccountToEdit: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => {
  return {
    onDeleteAccount: accountId => {
      dispatch(deleteAccount(accountId));
    },

    onSetAccountToEdit: account => {
      dispatch(setAccountToEdit(account));
    }
  };
};

export default withRouter(
  connect(null, mapDispatchToProps)(AccountItemContainer)
);
