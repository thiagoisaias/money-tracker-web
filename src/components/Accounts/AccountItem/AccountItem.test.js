import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AccountItem from "./AccountItem";

describe("AccountItem", () => {
  const mockProps = {
    id: "#UA12",
    name: "Mock Account",
    initialBalance: 999,
    currentBalance: 500.32,
    isActive: false,
    handleActiveItem: jest.fn()
  };

  const component = shallow(<AccountItem {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
