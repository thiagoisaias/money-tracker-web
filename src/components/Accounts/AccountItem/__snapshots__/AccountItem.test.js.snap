// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountItem renders properly 1`] = `
<styled.div
  isActive={false}
  onClick={[Function]}
>
  <styled.div>
    Mock Account
  </styled.div>
  <styled.div>
    $ 500.32
  </styled.div>
  <styled.div>
    <styled.div
      onClick={[Function]}
    >
      Edit
    </styled.div>
    <styled.div
      onClick={[Function]}
    >
      Delete
    </styled.div>
  </styled.div>
</styled.div>
`;
